src/main/java/com/ita/if103java/ims/service/impl/TransactionServiceImpl.java

```diff
@@ -34,53 +38,72 @@ public TransactionServiceImpl(TransactionDao transactionDao, AccountService acco
     }
 
     @Override
-    public TransactionDto findById(Long id) {
+    public TransactionDto findById(Long id, UserDetailsImpl userDetails) {
         final Transaction transaction = transactionDao.findById(id);
         return switch (transaction.getType()) {
-            case IN -> buildIncomeTransactionDto(transaction);
-            case OUT -> buildOutcomeTransactionDto(transaction);
-            case MOVE -> buildMoveTransactionDto(transaction);
+            case IN -> buildIncomeTransactionDto(transaction, userDetails);
+            case OUT -> buildOutcomeTransactionDto(transaction, userDetails);
+            case MOVE -> buildMoveTransactionDto(transaction, userDetails);
         };
     }
 
-    private TransactionDto buildIncomeTransactionDto(Transaction transaction) {
+    @Override
+    public Transaction create(ItemTransactionRequestDto itemTransactionRequestDto,
+                              User user,
+                              Long associateId,
+                              TransactionType type) {
+        final Transaction transaction = new Transaction();
+        transaction.setAccountId(user.getAccountId());
+        transaction.setAssociateId(associateId);
+        transaction.setItemId(itemTransactionRequestDto.getItemDto().getId());
+        transaction.setQuantity(itemTransactionRequestDto.getQuantity());
+        transaction.setWorkerId(user.getId());
+        transaction.setType(type);
+        switch (type) {
+            case OUT -> transaction.setMovedFrom(itemTransactionRequestDto.getSourceWarehouseId());
+            case IN -> transaction.setMovedTo(itemTransactionRequestDto.getDestinationWarehouseId());
+        }
+        return transactionDao.create(transaction);
```

Comment:
Similar to [ItemTransactionUtil](https://github.com/IF-103-Java/ims/blob/e7afdfa712ad29db1d4952902ea1e988d436d5e8/src/main/java/com/ita/if103java/ims/util/ItemTransactionUtil.java#L10)

Should be changed at new PR @Hvozdianchuck 
```java 
transactionDao.create(ItemTransactionUtil.createTransaction(.....))
```
onto 
```java
transactionService.create(....)
```